docs/DeveloperGuide.md

```diff
@@ -1,34 +1,845 @@
+		- [Design Considerations](#474-design-considerations)
+- [Testing](#5-testing)
+	- [Using IntelliJ JUnit Tests](#51-using-intellij-junit-tests)
+	- [Using Input-Output Tests](#52-using-input-output-tests)
+- [Appendices](#6-appendices)
+    - [Product Scope](#61-appendix-a-product-scope)
+    - [User Stories](#62-appendix-b-user-stories)
+    - [Use Cases](#63-appendix-c-use-cases)
+    - [Non-Functional Requirements](#64-appendex-d-non-functional-requirements)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 1. Setting up
+This section will guide you on how to set up this project on your own computer.
 
+### 1.1. Prerequisites
 
-## Product Scope
-### Target user profile
+1.  JDK 11 or above
 
-{Describe the target user profile}
+2.  IntelliJ IDE
 
-### Value proposition
+### 1.2. Setting up the project
+1.  Fork this repository, and clone the fork to your computer
 
-{Describe the value proposition: what problem does it solve?}
+2.  Open the IntelliJ IDE. If you are not in the welcome screen, click `File` &gt; `Close Project` to close the existing project.
 
-## User Stories
+3.  Set up the correct JDK version for Gradle
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+    1.  Click `Configure` &gt; `Project Defaults` &gt; `Project Structure`
 
-## Non-Functional Requirements
+    2.  Click `New...` and find the directory of the JDK
 
-{Give non-functional requirements}
+4.  Click on `Import Project`
 
-## Glossary
+5.  Locate and select the `build.gradle` file, then click `OK`
+
+6.  Click `Open as Project`
+
+7.  Click `OK` to use the default settings provided
+
+### 1.3. Verifying the Setup
+1.  In an IntelliJ terminal, run `gradlew build`
+
+2.  Move to the folder `build` &gt; `libs` and run: `java -jar atas-2.0.0.jar`
+
+    1.  To use **ATAS**, type a valid command into the terminal and press the enter key to run the command.  
+        e.g. Typing `help` and pressing the enter key will show the available commands
+
+    2.  Some example commands you can try to get familiar with **ATAS**:
+
+        -   `help`: Lists the commands that **ATAS** supports.
+
+        -   `assignment n/Assignment One m/CS2113T d/01/02/20 1600 c/Important Assignment`: Adds an assignment called **Assignment One** for the module **CS2113T**. This assignment is due on **1st February 2020 4pm** and the comments for this assignment is that it is an **Important Assignment**.
+
+        -   `exit`: Exits **ATAS**.
+
+## 2. Design
+This section will give a high-level overview of how various components in **ATAS** function and interact with each other.
```

Comment:
It is not very clear what your application is trying to achieve. Maybe can add a short sentence summarising what your app does?